.github/workflows/build_and_run_test_app.yml

```diff
@@ -1,13 +1,16 @@
 name: Build and Run Test Application
 
 on:
+  schedule:
+    - cron: '0 0 * * *' # Once per day at midnight
   pull_request:
     types: [opened, reopened, synchronize]
 
 jobs:
   build:
     name: Build Test App
     strategy:
+      fail-fast: false
```

Comment:
Not sure if this is possible or easy, but keeping fail-fast: true does make sense for PRs. For schedules indeed fail-fast: false is more useful. Perhaps `${{ github.event_name == 'pull_request' }}` would do the trick?

Reply:
I'll give it a try. Since we're talking about making it faster: Do we need nightly builds/tests again release 4.3 and 4.4 branches? Or is latest enough?

Reply:
I think as for the nightly build, time doesn't matter that much, since there's no developer eagerly waiting for their pipeline to pass. It's just a comment related to pull requests.

Reply:
I'd prefer leaving `fail-fast: false` for all cases. It will help contributors see whether the compilation/test issues is global, or specific to certain IDF version.

Reply:
Okay, let's keep it this way